os/kernel/semaphore/sem_holder.c

```diff
@@ -908,11 +908,8 @@ void sem_releaseholder(FAR sem_t *sem, FAR struct tcb_s *rtcb)
 		 */
 		pholder->counts--;
 
-#ifndef CONFIG_PRIORITY_INHERITANCE
-		if (pholder->counts <= 0) {
-			sem_freeholder(sem, pholder);
-		}
-#endif
+		/* Remove the holder from the holder list */
```

Comment:
Can we check if the holder thread is active or not with :  sched_verifytcb(htcb) 
If the thread is not active then we can sem_freeholder in else case?